src/offset/local/windows.rs

```diff
@@ -8,271 +8,105 @@
 }
 
 /// Converts a local `NaiveDateTime` to the `time::Timespec`.
 pub(super) fn naive_to_local(d: &NaiveDateTime, local: bool) -> LocalResult<DateTime<Local>> {
-    let tm = Tm {
-        tm_sec: d.second() as i32,
-        tm_min: d.minute() as i32,
-        tm_hour: d.hour() as i32,
-        tm_mday: d.day() as i32,
-        tm_mon: d.month0() as i32, // yes, C is that strange...
-        tm_year: d.year() - 1900,  // this doesn't underflow, we know that d is `NaiveDateTime`.
-        tm_wday: 0,                // to_local ignores this
-        tm_yday: 0,                // and this
-        tm_isdst: -1,
-        // This seems pretty fake?
-        tm_utcoff: if local { 1 } else { 0 },
-        // do not set this, OS APIs are heavily inconsistent in terms of leap second handling
-        tm_nsec: 0,
-    };
+    let naive_sys_time = WinSystemTime::from_naive_datetime(d);
 
-    let spec = Timespec {
-        sec: match local {
-            false => utc_tm_to_time(&tm),
-            true => local_tm_to_time(&tm),
-        },
-        nsec: tm.tm_nsec,
+    let local_sys_time = match local {
+        false => LocalSysTime::from_utc_time(naive_sys_time),
+        true => LocalSysTime::from_local_time(naive_sys_time),
     };
 
-    // Adjust for leap seconds
-    let mut tm = spec.local();
-    assert_eq!(tm.tm_nsec, 0);
-    tm.tm_nsec = d.nanosecond() as i32;
-
-    // #TODO - there should be ambiguous cases, investigate?
-    LocalResult::Single(tm_to_datetime(tm))
-}
+    if let Ok(mut local) = local_sys_time {
+        assert_eq!(local.nsecs, 0);
+        local.set_nsecs(d.nanosecond() as i32);
 
-/// Converts a `time::Tm` struct into the timezone-aware `DateTime`.
-fn tm_to_datetime(mut tm: Tm) -> DateTime<Local> {
-    if tm.tm_sec >= 60 {
-        tm.tm_nsec += (tm.tm_sec - 59) * 1_000_000_000;
-        tm.tm_sec = 59;
+        return LocalResult::Single(local.datetime());
     }
 
-    let date = NaiveDate::from_ymd_opt(tm.tm_year + 1900, tm.tm_mon as u32 + 1, tm.tm_mday as u32)
-        .unwrap();
-    let time = NaiveTime::from_hms_nano(
-        tm.tm_hour as u32,
-        tm.tm_min as u32,
-        tm.tm_sec as u32,
-        tm.tm_nsec as u32,
-    );
-
-    let offset = FixedOffset::east_opt(tm.tm_utcoff).unwrap();
-    DateTime::from_utc(date.and_time(time) - offset, offset)
+    LocalResult::None
 }
 
-/// A record specifying a time value in seconds and nanoseconds, where
-/// nanoseconds represent the offset from the given second.
-///
-/// For example a timespec of 1.2 seconds after the beginning of the epoch would
-/// be represented as {sec: 1, nsec: 200000000}.
-struct Timespec {
+struct InnerDuration {
     sec: i64,
     nsec: i32,
 }
 
-impl Timespec {
+impl InnerDuration {
+    /// Constructs a new LocalSysTime representing the current time in UTC
     /// Constructs a timespec representing the current time in UTC.
-    fn now() -> Timespec {
+    fn now_locally() -> LocalSysTime {
```

Comment:
It seems like there should be Windows API that would let us get both the current `WinSystemTime` and the current timezone offset in one system call, instead of doing this stuff?